cli2/cli.ts

```diff
@@ -0,0 +1,156 @@
+    const parsedMorphirJson = JSON.parse(morphirJsonContent.toString())
+
+    // All the files in the src directory are read
+    const sourcedFiles = await readElmSourceFiles(path.join(projectDir, parsedMorphirJson.sourceDirectory))
+
+    //To check if the morphir-ir.json already exists
+    const morphirIrPath: string = path.join(projectDir, 'morphir-ir.json')
+    try {
+        await accessFile(morphirIrPath, fs.constants.F_OK)
+        console.log(`${morphirIrPath}, will be passed to the worker along side changed to update the ir`)
+    } catch (err) {
+        console.log(`${err}, not found`)
+        return packageDefinitionFromSource(parsedMorphirJson, sourcedFiles, options)
+    }
+    // return packageDefinitionFromSource(parsedMorphirJson, sourcedFiles, options)
+}
+
+//generating a hash with md5 algorithm for the content of the read file
+const createHashFile = (contentOfFile: any) => {
+    let hash = crypto.createHash('md5');
+    let data = hash.update(contentOfFile, 'utf-8')
+    let gen_hash = data.digest('hex')
+    return gen_hash;
+}
+
+async function packageDefinitionFromSource(parsedMorphirJson :any, sourcedFiles :any, options:any, ) {
+    return new Promise((resolve, reject) => {
+        worker.ports.jsonDecodeError.subscribe((err: any) => {
+            reject(err)
+        })
+
+        worker.ports.packageDefinitionFromSourceResult.subscribe(([err, ok]: any) => {
+            if (err) {
+                reject(err)
+            } else {
+                resolve(ok)
+            }
+        })
+
+        const opts = {
+            typesOnly: options.typesOnly
+        }
+
+        worker.ports.packageDefinitionFromSource.send([opts, parsedMorphirJson, sourcedFiles])
+    })
+}
+
+async function differenceInPathAndHash(hashFilePath: string, filePath: string, fileChange: string,
+    fileHash: Map<string, string>, fileChangesMap: Map<string, Array<string>>) {
+    const readHashFile = await readFile(hashFilePath)
+    let hashJson = JSON.parse(readHashFile.toString())
+    for (let key in hashJson) {
+        fileHash.set(key, hashJson[key])
+    }
+    const readContent = await readFile(filePath)
+    const hash = createHashFile(readContent)
+    if (fileHash.has(filePath)) {
+        if (fileHash.get(filePath)) {
+            if (fileHash.get(filePath) !== hash) {
+                fileHash.set(filePath, hash)
+                fileChangesMap.set(filePath, ['Updated', readContent.toString()])
+            }
+        }
+    }
+    else {
+        fileHash.set(filePath, hash)
+        fileChangesMap.set(filePath, ['Insert', readContent.toString()])
+    }
+
+    let fileChangeObject = Object.fromEntries(fileChangesMap)
+    await fsWriteFile(fileChange, JSON.stringify(fileChangeObject, null, 2))
+
+    let jsonObject = Object.fromEntries(fileHash)
+    await fsWriteFile(hashFilePath, JSON.stringify(jsonObject, null, 2))
+
+    const fileChangeJson = await readFile(fileChange)
+    return fileChangeJson
```

Comment:
Instead of saving and reading the file changes you should just return them. We don't need to store the file changes.